build/room_ensurer/room_ensurer.py

```diff
@@ -143,60 +143,65 @@ def _ensure_room_for_network(self, network: Networks, alias_fragment: str) -> No
             server_name: self._get_room(server_name, room_alias_prefix)
             for server_name in self._known_servers.keys()
         }
-        first_server_room_info = room_infos[self._first_server_name]
-
-        if not first_server_room_info:
-            log.warning("First server room missing")
-            if self._is_first_server:
-                log.info("Creating room", server_name=self._own_server_name)
-                first_server_room_info = self._create_room(room_alias_prefix)
-                room_infos[self._first_server_name] = first_server_room_info
-            else:
-                raise EnsurerError("First server room missing.")
-
-        are_all_rooms_the_same = all(
-            room_info is not None and room_info.room_id == first_server_room_info.room_id
+
+        # if it is the first server in the list and no room is found, create room
+        if self._is_first_server and room_infos[self._own_server_name] is None:
+            log.info("Creating room", server_name=self._own_server_name)
+            first_server_room_info = self._create_room(room_alias_prefix)
+            room_infos[self._first_server_name] = first_server_room_info
+            log.info(
+                "Room created. Waiting for other servers to join.",
+                room_aliases=first_server_room_info.aliases,
+                room_id=first_server_room_info.room_id,
+            )
+            return
+
+        ensure_room_info = None
```

Comment:
I think this variable could be a bit clearer, although it is understandable in context. Only it's `room_id` attribute is ever used and the variable name is not self explanatory to me. Should we use something like `expected_room_id`, instead?

Reply:
We use the class [`RoomInfo`](https://github.com/raiden-network/raiden-service-bundle/blob/24a7260618441600944106cae4cd01f08688fdeb/build/room_ensurer/room_ensurer.py#L84) to gather aliases, room_id and the corresponding server_name. Aliases and server name are mostly used for logging purposes. The function `get_room` fetches the room information from a server and packs it in this object. Even though we only use the room_id in `ensure_rooms` I dont think it is necessary to not reuse the function `get_room()` and directly call the API. 

I think `ensure_room_info` is a pretty clear naming under consideration that the corresponding class of the option is called `RoomInfo`. Do you have a better idea? (expected_room_id is wrong in this case)

Reply:
maybe `expected_room_info` ?

Reply:
Now I see some usage of the other attributes in `_ensure_admin_power_levels`, although just for logging. I missed that part before.

> Even though we only use the room_id in ensure_rooms I dont think it is necessary to not reuse the function get_room() and directly call the API.

Certainly not, but `get_room().room_id` could be used.`

> maybe `expected_room_info` ?

With the full RoomInfo that's slightly misleading, since we don't expect the same `server_name`. In that case it is hard to find a name, since the concept is "a room where I expect all other rooms with the same alias to share the room_id". The actual algorithm would be clearer with just the room_id, but at the expense of removing a way to easily log the alias.

Do as you see fit.